src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -576,31 +592,74 @@ private static String getStringInsideTag(String result, DetailNode detailNode) {
     }
 
     /**
-     * Finds and returns first sentence.
+     * Finds and returns the first sentence.
      *
-     * @param ast Javadoc root node.
-     * @return first sentence.
+     * @param ast The Javadoc root node.
+     * @param period The configured period symbol.
+     * @return The first sentence up to and excluding the period, if a sentence ending was found.
      */
-    private static String getFirstSentence(DetailNode ast) {
-        final StringBuilder result = new StringBuilder(256);
-        final String periodSuffix = DEFAULT_PERIOD + ' ';
-        for (DetailNode child : ast.getChildren()) {
-            final String text;
-            if (child.getChildren().length == 0) {
-                text = child.getText();
+    private static Optional<String> getFirstSentence(DetailNode ast, String period) {
```

Comment:
We should return Optional, and do `if empty`

Reply:
I can't find a better way to combine the optional with the break statement in `getFirstSentence`.

If I try to use `.isPresent` or `.isEmpty` combined with a `.get`, it tells me to use `.orElseThrow` instead. That looks confusing, because it implies there's some way the optional could still be empty after the isPresent/isEmpty check.

We can't use `.ifPresent` or `.ifPresentOrElse` because you can't put the break statement inside the lambda.

Can you think of another way to do it? I think I'd be fine with leaving it as `.orElse(null)`.

Reply:
I've made it use `.ifPresentOrElse` in `validateUntaggedSummary`.

Reply:
> We should return Optional, and do if empty

@romani why?

Reply:
I've just pushed a version with the nullable annotation.

Reply:
We did not need to fallback to null.
But let's not block this PR by this, we can refactor in separate PR.

Reply:
https://github.com/checkstyle/checkstyle/issues/14881